### PR TITLE
Fixed metadata names

### PR DESCRIPTION
**Pull request by g4s8**
#29 - Changed `Repomd` implementation to generate metadata file names **before** updating `repomd.xml`.

### CODE REVIEW

**Comment by paulodamaso** [comment]
@g4s8 PLease assign to me, I'm available

**Comment by 0crat** [comment]
This pull request [#39](https://github.com/artipie/rpm-adapter/issues/39) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/f2993b84-bdef-4fdd-b2e3-9cf85b7df5e7); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/rpm/Repomd.java [hunk 1]
```diff
@@ -119,71 +119,95 @@ private Completable loadRepomd(final Path file) {
      * Perform the actual update.
      *
      * @param type The file type
-     * @param act The action to perform
-     * @param temp The temp
+     * @param act Action
+     * @param repomd The temp
```
Comment (paulodamaso) [inline src/main/java/com/artipie/rpm/Repomd.java]: @g4s8 These paremeters could be better described, it is almost impossible to understand them without reading the entire method code

File: src/main/java/com/artipie/rpm/Repomd.java [hunk 2]
```diff
@@ -119,71 +119,95 @@ private Completable loadRepomd(final Path file) {
...
-                                            )
-                                        )
-                                        .andThen(new Update(temp).apply(rxml))
-                                        .andThen(
-                                            Completable.fromAction(
-                                                () -> new BlockingStorage(this.storage).save(
-                                                    new Key.From("repodata/repomd.xml"),
-                                                    Files.readAllBytes(temp)
-                                                )
+        return Single.fromCallable(() -> Files.createTempFile(type, ".xml"))
```
Comment (paulodamaso) [inline src/main/java/com/artipie/rpm/Repomd.java]: @g4s8 I think that you could leave an issue to fix this methods / architecture: currently both `performUpdate` methods do a lot of thinks (creating files, zipping contents, saving file) and I think that these behaviors should be split into several methods

**Comment by g4s8** [comment]
@paulodamaso I've added a puzzle to address you comments. Will merge this PR, since it's urgent a bit.

**Comment by paulodamaso** [comment]
@g4s8 Good to me, merge please

**Comment by 0crat** [comment]
Job was finished in 21 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Pull request merged**